ihatemoney/web.py

```diff
@@ -898,6 +901,50 @@ def statistics():
     )
 
 
+def build_etag(project_id, last_modified):
+    return hashlib.md5(
+        (current_app.config["SECRET_KEY"] + project_id + last_modified).encode()
+    ).hexdigest()
+
+
+@main.route("/<project_id>/feed/<string:token>.xml")
+def feed(token):
+    verified_project_id = Project.verify_token(
+        token, token_type="feed", project_id=g.project.id
+    )
+    if verified_project_id != g.project.id:
+        abort(404)
+
+    weighted_bills = g.project.get_bill_weights_ordered().paginate(
+        per_page=100, error_out=True
+    )
+
+    bills_last_modified = [
+        bill.versions[0].transaction.issued_at for bill in g.project.get_bills()
+    ]
+    project_last_modified = g.project.versions[0].transaction.issued_at
```

Comment:
Add a comment mentioning SQLAlchemy-Continuum, because it's not obvious from the code (and same thing above). If we ever decide to remove, upgrade or replace it, it will be easier to identify this particular usage.

By the way, will your code still work if a project has history disabled?

Reply:
> Add a comment mentioning SQLAlchemy-Continuum, because it's not obvious from the code (and same thing above). If we ever decide to remove, upgrade or replace it, it will be easier to identify this particular usage.

Done.

> By the way, will your code still work if a project has history disabled?

I added a test that checks that.

Reply:
Looks good, thanks!